fossunited/tests/utils.py

```diff
@@ -325,3 +328,123 @@ def insert_rsvp_submission(linked_rsvp: str, **kwargs):
 
     except Exception:
         raise
+
+
+def insert_test_hackathon(**kwargs):
+    """
+    Generate a test hackathon with flexible configuration options.
+
+    Args:
+        permalink (str, optional): Permalink of the hackathon. Defaults to random slug.
+        hackathon_name (str, optional): Name of the hackathon. Defaults to random text.
+        hackathon_type (str, optional): Type of the hackathon. Defaults to "Hybrid".
+        start_date (datetime, optional): Hackathon start date. Defaults to tomorrow.
+        end_date (datetime, optional): Hackathon end date. Defaults to day after tomorrow.
+        hackathon_description(str, optional):Description of hackathon. Defaults to "Test Hackathon"
+        **kwargs: Additional arguments to be passed to the hackathon document.
+
+    Returns:
+        hackathon doc: Created hackathon document
+
+    Raises:
+        Exception: For any unexpected errors during hackathon generation
+    """
+    try:
+        hackathon_data = {
+            "doctype": HACKATHON,
+            "permalink": kwargs.get("permalink", fake.slug().replace("-", "_")),
+            "hackathon_name": kwargs.get("hackathon_name", fake.text(max_nb_chars=20).strip()),
+            "hackathon_type": kwargs.get("hackathon_type", "Hybrid"),
+            "start_date": kwargs.get("start_date", datetime.today() + timedelta(days=1)),
+            "end_date": kwargs.get("end_date", datetime.today() + timedelta(days=2)),
+            "hackathon_description": kwargs.get("hackathon_description", "Test Hackathon"),
+        }
+
+        for key, value in kwargs.items():
+            if key not in hackathon_data:
+                hackathon_data[key] = value
+
+        hackathon = frappe.get_doc(hackathon_data)
+        hackathon.insert()
+        hackathon.reload()
+
+        return hackathon
+
+    except Exception:
+        raise
```

Comment:
not sure why we're doing this - the function will behave exactly this way if we complete remove the `try/except` block. Earlier, we were logging errors so it sort of made sense but i'm not sure it makes sense now - the function will fail loudly so the log messages feel redundant.

what was the intention behind the `try/except` block? this code is meant purely for testing so it's okay to fail fast/clearly.

Reply:
Just tried to follow the other util functions, but i guess we can lose the try block here. Removed